codegen/generators/python.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+module Generator
+  class Python < Base
+    def format_enum_value(value)
+      value.downcase.gsub(/[.\/+\s-]/, '_')
+    end
+
+    def get_sorted_properties(definition)
+      required_fields = definition['required'] || []
+      definition['properties'].sort_by do |name, *|
+        [required_fields.include?(name) ? 0 : 1, name]
+      end
+    end
+
+    def format_property(parent_type_name, property_name, property, required_fields)
+      snake_name = property_name.gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
+                               .gsub(/([a-z\d])([A-Z])/, '\1_\2')
+                               .downcase
+
+      property_type = get_property_type(parent_type_name, property_name, property)
+      is_required = required_fields.include?(property_name)
+
+      if is_required
+        "#{snake_name}: #{property_type}"
+      else
+        "#{snake_name}: Optional[#{property_type}] = None"
+      end
+    end
+
+    def get_property_type(parent_type_name, property_name, property)
+      type = type_for(parent_type_name, property_name, property)
+
+      if type.start_with?('list[')
+        list_type = type.match(/list\[(.*?)\]/)
```

Comment:
Whilst this isn't wrong / buggy. Is there anything slightly more nuanced we can use for the inner capture. Maybe `(\w+)` ?

Reply:
Also you can optimise this as you're only using the regexp matcher briefly

```
type[/list\[(.*?)\]/]
```

This will not store it in a variable, but it will store it in Regexp.last_match -> The index you want is `1` so then in the statements below you can call `Regexp.last_match(1)`

Reply:
Isn't there a better way rather than doing regexp match? I don't know ruby idioms, but in python I would return a boxed object that wraps the subtype, like `ArrayMarker(foo)`, and do an isinstwnce check here.

Reply:
I am not sure what you are trying to do here as it looks quite complicated.

Basically storing any regex details inside a `#match` object is the heaviest footprint. And storing things briefly using Regexp.last_match is a "tiny" bit lighter, and Ideally just using `#match?` is the best because this doesn't store anything

Reply:
If it's easier and you want me to tackle this as a follow-up can you make an issue and link to this given the size of this PR is quite large and first iteration is always the hardest one.